substrate/frame/examples/basic/src/tests.rs

```diff
@@ -81,10 +81,10 @@ impl pallet_balances::Config for Test {
 	type ExistentialDeposit = ConstU64<1>;
 	type AccountStore = System;
 	type WeightInfo = ();
-	type FreezeIdentifier = ();
-	type MaxFreezes = ();
-	type RuntimeHoldReason = ();
-	type RuntimeFreezeReason = ();
+	type FreezeIdentifier = RuntimeFreezeReason;
+	type MaxFreezes = VariantCountOf<RuntimeFreezeReason>;
```

Comment:
Can't we compute it internally using the already provided `RuntimeFreezeReason`? Do we still need to keep a separate config parameter?

Reply:
> https://github.com/paritytech/polkadot-sdk/issues/2997 (remove MaxFreezes and FreezeIdentifier)

I just saw this 🙈 

Shouldn't we just to go to that one directly?

Reply:
yes, exactly, that is the point of https://github.com/paritytech/polkadot-sdk/issues/2997 and will be done as separate PR

Reply:
> > #2997 (remove MaxFreezes and FreezeIdentifier)
> 
> I just saw this 🙈
> 
> Shouldn't we just to go to that one directly?

Well, I would better do it as a separate PR (needs migration from FreezeIdentifier to RuntimeFreezeReason or whatever , plus some checks and so on...).
And I am also going to address Kian's comment here: https://github.com/paritytech/polkadot-sdk/pull/4615#discussion_r1618574118, so it will be a big PR by itself.

This PR fixes correct usage of these freeze stuff, so it will be a better ground for that #2997 refactor.